python/xoscar/collective/test/test_pygloo_tcp_store.py

```diff
@@ -0,0 +1,622 @@
+import multiprocessing as mp
+import os
+import platform
+import shutil
+import time
+
+import numpy as np
+
+from ...tests.core import require_unix
+
+fileStore_path = "./collective"
+system_name = platform.system()
+
+
+def worker_allgather(rank):
+    from ..gloo import xoscar_pygloo as xp
+    from ..rendezvous import xoscar_store as xs
+
+    if rank == 0:
+        if os.path.exists(fileStore_path):
+            shutil.rmtree(fileStore_path)
+        os.makedirs(fileStore_path)
+    else:
+        time.sleep(0.5)
+
+    context = xp.rendezvous.Context(rank, 2)
+
+    if system_name == "Linux":
+        attr = xp.transport.tcp.attr("localhost")
+        dev = xp.transport.tcp.CreateDevice(attr)
+    else:
+        attr = xp.transport.uv.attr("localhost")
+        dev = xp.transport.uv.CreateDevice(attr)
+    opt = xs.TCPStoreOptions()
+    opt.port = 25001
+    opt.numWorkers = 2
+    if rank == 0:
+        opt.isServer = True
+    else:
+        opt.isServer = False
+
+    store = xs.TCPStore("127.0.0.1", opt)
+    custom_store = xp.rendezvous.CustomStore(store)
+    store = xp.rendezvous.PrefixStore(str(2), custom_store)
+
+    context.connectFullMesh(store, dev)
+
+    sendbuf = np.array([[1, 2, 3], [1, 2, 3]], dtype=np.float32)
+    recvbuf = np.zeros([2] + list(sendbuf.shape), dtype=np.float32)
+    sendptr = sendbuf.ctypes.data
+    recvptr = recvbuf.ctypes.data
+
+    assert sendbuf.size * 2 == recvbuf.size
+
+    data_size = (
+        sendbuf.size if isinstance(sendbuf, np.ndarray) else sendbuf.numpy().size
+    )
+    datatype = xp.glooDataType_t.glooFloat32
+
+    xp.allgather(context, sendptr, recvptr, data_size, datatype)
+
+    print(f"rank {rank} sends {sendbuf},\nreceives {recvbuf}")
```

Comment:
add an assert, not print.

Reply:
ok, I have changed it to `np.testing.assert_array_equal`